File: OP_laba5_sharp/OP_laba5_sharp/Line.cs
using System;

namespace OP_laba5_sharp
{
    public class Line
    {
        protected Dot _start;
        protected Dot _end;
        protected double dx;
        protected double dy;
    
        
        public Line()
        {
            this._start = null;
            this._end = null;
            dx = 0;
            dy = 0;
        }
        
        public Line(Dot start, Dot end)
        {
            this._start = start;
            this._end = end;
            dx = _end.X - _start.X;
            dy = _end.Y - _start.Y;
        }

        public double GetLength()
        {
            return Math.Sqrt(dx * dx + dy * dy);
        }
        
        
        
    }
}

File: OP_laba5_c++/src/Line.cpp
#include "Line.h"

Line::Line()
{
    start = Dot(0.0,0.0);
    finish = Dot(0.0, 0.0);
    dx = 0.0;
    dy = 0.0;
}

Line::Line(Dot first, Dot second)
{
    start = first;
    finish = second;
    dx = start.x - finish.x;
    dy = start.y - finish.y;
}

double Line::GetLength()
{
    return sqrt(dx*dx - dy*dy);
}
